backend/src/auth/jwt-auth.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class JwtAuthGuard extends AuthGuard('jwt') {
+  canActivate(context: ExecutionContext): boolean | Promise<boolean> | Observable<boolean> {
+    const whitelistedPaths = ['/api/oauth2', '/api/logout'];
```

Comment:
**chore:** I don't think we need to whitelist the logout path as we don't redirect to it